core/common/src/main/java/alluxio/collections/ConcurrentHashSet.java

```diff
@@ -104,6 +104,9 @@ public boolean equals(Object o) {
       return true;
     }
     // allow equality with any Set.
+    if (!(o instanceof ConcurrentHashSet)) {
```

Comment:
But shouldn't two implementations of `Set` be considered equal if all of their elements are equal?